Document/ivyportal 16878 new statistics documentation & g UI test

lmluat-axonivy opened the pull request:
- Update user guide
- Update dev guide

mnhnam-axonivy commented on Documentation/portal-guide/source/portal-developer-guide/dashboard/new-dashboard-statistic-widget.rst:
```diff
@@ -13,68 +13,70 @@ Below is a sample JSON definition of a statistics widget in the Portal dashboard
-               "locale": "es",
-               "value": "Tareas por prioridad"
-            }
+      "id": "1",
```
This is JSON example of Portal.StatisticCharts not statistic widget on the dashboard.
Example for statistic widget on the dashboard

`{
          "type": "client-statistic",
          "id": "client_statistic_1",
          "layout": {
            "w": 2,
            "h": 2,
            "x": 8,
            "y": 2
          },
          "chartId": "10"
        }`

mnhnam-axonivy commented on Documentation/portal-guide/source/portal-developer-guide/settings/index.rst:
```diff
@@ -437,60 +437,91 @@ Data model:
 ^^^^^^^^^^^^^^^^^^^^^^^
-You can define the standard statistic charts via the following JSON file. They
-will be shown as the default charts on the statistic page.
+You can define the standard statistic widgets via the following JSON file. They
```
Change to:

You can define filter logic, appearance, and other settings for all statistic charts that can be use by the statistic widget of Portal dashboard.

the term "statistic widget" -> link to the Statistic widget for Dashboard page.

mnhnam-axonivy commented on Documentation/portal-guide/source/portal-developer-guide/dashboard/new-dashboard-statistic-widget.rst:
```diff
@@ -13,68 +13,70 @@ Below is a sample JSON definition of a statistics widget in the Portal dashboard
+   ``filter``: list of filters to apply for each chart. Please refer to :ref:`Portal.StatisticCharts <portal-statistic-charts>` for more information.
 
-      ``style`` (optional): add inline style to HTML DOM of the widget
+   ``chartTarget``: the chart target of the widget, ``TASK`` or ``CASE``.
```
the entity type that you want to use as the target of the chart. We are supporting the main entity types of the ivy workflow:
  - ``TASK``: Ivy Task
  - ``CASE``: Ivy Case

mnhnam-axonivy commented on Documentation/portal-guide/source/portal-developer-guide/dashboard/new-dashboard-statistic-widget.rst:
```diff
@@ -13,68 +13,70 @@ Below is a sample JSON definition of a statistics widget in the Portal dashboard
+   ``chartTarget``: the chart target of the widget, ``TASK`` or ``CASE``.
 
-   ``chart``: the chart data model of the widget
+   ``chartType``: type of the chart such as ``Pie``, ``Bar``, ``Line`` and ``KPI``. For ``KPI`` chart, you have to set the type as ``Number`` instead of ``KPI``.
```
Chart type that you want to show on the UI. There are 4 chart types:
  - ``Pie``: Pie chart
  (Pie chart image)
  - ``Bar``: Bar chart
  (Bar chart image)
  - ``Line``: Line chart
  (Line chart image)
  - ``Number``: Labelled number chart
  ( KPI chart image)

mnhnam-axonivy commented on Documentation/portal-guide/source/portal-developer-guide/dashboard/new-dashboard-statistic-widget.rst:
```diff
@@ -13,68 +13,70 @@ Below is a sample JSON definition of a statistics widget in the Portal dashboard
-      ``ELAPSED_TIME_BY_CASE_CATEGORY``, and ``CASES_BY_CATEGORY``
+   ``icon``: the icon of each widget.
+   
+   ``refreshInterval``: statistic widget refresh interval in seconds 
```
Statistic chart refresh interval in seconds.

mnhnam-axonivy commented on Documentation/portal-guide/source/portal-developer-guide/dashboard/new-dashboard-statistic-widget.rst:
```diff
@@ -13,68 +13,70 @@ Below is a sample JSON definition of a statistics widget in the Portal dashboard
+   ``refreshInterval``: statistic widget refresh interval in seconds 
 
-      ``filter``: list of filters to apply for each chart. Please refer to :ref:`Portal.StatisticCharts <portal-statistic-charts>` for more information.
+For a specific chart such as ``Bar``, ``Line`` or ``KPI`` chart, it requires some additional and required fields. Please refer to :ref:`Portal.StatisticCharts <portal-statistic-charts>` for more information.
```
Could we change the name``KPI`` to ``Number``?

phochstr commented on Documentation/portal-guide/source/portal-user-guide/statistic-chart/index.rst:
```diff
@@ -24,50 +24,55 @@ Available Standard Charts
 
-    - Displays number of tasks set to expire on the current day 
-    - Chart type: :bdg-info:`KPI`
+    - This chart shows the average time of cases by category. 
```
time --> processing time / elapsed time

time alone does not say enough. You may somehow collect the actual duration of the work on the cases, or the time elapsed between start of the case and its being terminated - completed or destroyed, i.e., its elapsed time.

phochstr commented on Documentation/portal-guide/source/portal-user-guide/statistic-chart/index.rst:
```diff
@@ -24,50 +24,55 @@ Available Standard Charts
 
-    - Displays number of tasks expiring by the end of the current week.
-    - Chart type: :bdg-info:`KPI`
+    - This chart shows all started cases per day within the last 5 days.
```
Does this chart show the cases, or just the number of cases, started each day? I'd assume the latter.

--> This chart shows the number of cases started per day for the last 5 days.

phochstr commented on Documentation/portal-guide/source/portal-user-guide/statistic-chart/index.rst:
```diff
@@ -24,50 +24,55 @@ Available Standard Charts
 |dashboard-statistic-widget-demo|
 
-- Tasks by priority
+- Tasks by Priority
```
Standardization: Either Capital Case Titles, or Initial capital case titles, but don't Do something In between.

Top Priority: 3 days 
--> Top Priority: 3 Days or 
     Top priority: 3 days

for all following example titles.

phochstr reviewed (commented):
Lots of details - mainly singular/plural, temporal vs. value-based and consistency/standardization  issues.

Despite all of the comments, very nice additions to the doc!

lmluat-axonivy commented on Documentation/portal-guide/source/portal-developer-guide/settings/index.rst:
```diff
@@ -437,60 +437,112 @@ Data model:
+   - ``yTitles``: the multilingual display title for the y-axis
 
--  ``position``: position index of chart. Used to order the charts in the UI
+- ``numberChartConfig``: additional fields for configuring the ``Number`` chart, you could add when chart type is ``number``
```
the chart type should be lower case in JSON file so I use number instead of Number

phochstr replied:
OIC. OK.

Then I don't see why you InitialCap this:

`chartType``: type of the chart such as ``Pie``, ``Bar``, ``Line`` and ``KPI``. For ``KPI`` chart, you have to set the type as ``Number`` instead of ``KPI``.

The chartType is an entry in the JSON, so the values should be lowercase, too, right?
BTW: The last sentence above seems to be unnecessary - you named the KPI chart type Number, so it is no longer necessary to explain the type name switch.

lmluat-axonivy replied:
I already removed the line that you mention.

lmluat-axonivy commented:
Thank for your review @phochstr . I have updated both user and developer guide. In user guide, I added images for each chart and changed the name following the standard.
